pkg/controller/snapshot_controller_test.go

```diff
@@ -0,0 +1,91 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controller
+
+import (
+	crdv1 "github.com/kubernetes-csi/external-snapshotter/pkg/apis/volumesnapshot/v1alpha1"
+	"k8s.io/client-go/tools/cache"
+	"testing"
+)
+
+func storeVersion(t *testing.T, prefix string, c cache.Store, version string, expectedReturn bool) {
+	content := newContent("contentName", classEmpty, "sid1-1", "vuid1-1", "volume1-1", "snapuid1-1", "snap1-1", nil, nil)
+	content.ResourceVersion = version
+	ret, err := storeObjectUpdate(c, content, "content")
+	if err != nil {
+		t.Errorf("%s: expected storeObjectUpdate to succeed, got: %v", prefix, err)
+	}
+	if expectedReturn != ret {
+		t.Errorf("%s: expected storeObjectUpdate to return %v, got: %v", prefix, expectedReturn, ret)
+	}
+
+	// find the stored version
+
+	contentObj, found, err := c.GetByKey("contentName")
+	if err != nil {
+		t.Errorf("expected content 'contentName' in the cache, got error instead: %v", err)
+	}
+	if !found {
+		t.Errorf("expected content 'contentName' in the cache but it was not found")
+	}
+	content, ok := contentObj.(*crdv1.VolumeSnapshotContent)
+	if !ok {
+		t.Errorf("expected content in the cache, got different object instead: %#v", contentObj)
+	}
+
+	if ret {
+		if content.ResourceVersion != version {
+			t.Errorf("expected content with version %s in the cache, got %s instead", version, content.ResourceVersion)
+		}
+	} else {
+		if content.ResourceVersion == version {
+			t.Errorf("expected content with version other than %s in the cache, got %s instead", version, content.ResourceVersion)
```

Comment:
we probably should be using a golang test framework like testify. Writing `t.Errorf(..)` every time is a PAIN! Nice job though.